CDS/scripts/png_to_jpeg.rb

```diff
@@ -0,0 +1,37 @@
+#!/usr/bin/env ruby
+
+#This module tests if an image file is a PNG and, if it is, attempts to convert it into a JPEG.
+#It depends on ImageMagick being available on the local node (apt-get install ImageMagick or yum install ImageMagick or port install ImageMagick etc.), specifically the 'convert' and 'identify' commands.
+#
+#Arguments:
+# <input_key>{meta:FileNameKey} - use the provided key as an input
+# <output_key>keyname - output the path of the file into this key in the meta: section of the datastore
+
+#END DOC
+
+require 'CDS/Datastore'
+
+#START MAIN
+store=Datastore.new('png_to_jpeg')
+
+input_image=store.substitute_string(ENV['input_key'])
+
+cmdline="identify -format '%m' '#{input_image}'"
+
+command_result = system(cmdline)
+
+if (command_result)
+  cmdline="convert '#{input_image}' '#{input_image}'"
```

Comment:
won't this over-write the original filename? shouldn't we make an output filename by swapping .png for .jpg and using that as the output?

Reply:
It will. This is meant to work like this.

No. Because you cannot have conditionals in CDS route files I have built it like this. You feed in a PNG or a JPEG with a file name such as 'CAPI.jpg' and the script converts the file into a real JPEG if it is actually a PNG.

Reply:
OK, I don't like overwriting the file but I guess it's valid in this case.